script.js

```diff
@@ -1,406 +1,292 @@
+const tempToggle = () => {
+  lowToHigh = !lowToHigh
+  sortByTemp()
+}
+
+//Function that sorts high to low and low to high depending of value atm
+const sortByTemp = () => {
+  if (lowToHigh) {
+  destinations.sort((destA, destB) => {
+      return destB.temperature - destA.temperature
+  })
+  } else {
+    destinations.sort((destA, destB) => {
+      return destA.temperature - destB.temperature
+    })
+}
+destinationDiv.classList.replace('filtered-destinations', 'load-destinations')
+loadDestinations(destinations)
+}
+
+//function that shift the value each time button is clicked, to enable toggle
+const rateToggle = () => {
+  lowToHigh = !lowToHigh
+  sortByRate()
+}
+
+// Function that sort by rate, ascending or descending depending on value atm
+const sortByRate = () => {
+  if (lowToHigh) {
+    destinations.sort((destA, destB) => {
+      return destB.rating - destA.rating
+   })
+  } else {
+    destinations.sort((destA, destB) => {
+      return destA.rating - destB.rating
+    })
   }
-]
+  destinationDiv.classList.replace('filtered-destinations', 'load-destinations')
+  loadDestinations(destinations)
+} 
+
+//Function for filter dropdown menu with conditional statement for all countries
+const filterCountry = () => {
+const countryValue = dropDown.value
+  if (countryValue === 'all'){
+    loadDestinations(destinations)
+    destinationDiv.classList.replace('filtered-destinations', 'load-destinations')
+    return
+  }
+  const filteredList = destinations.filter((dest) => {
+    const matchesType = dest.country === countryValue
+    return matchesType 
+  })
+  destinationDiv.classList.replace('load-destinations', 'filtered-destinations')
+  loadDestinations(filteredList)
+}
+
+//Function for random-button, together with delayed text bubble 
+const filterRandom = () => {
+  let randomDestination = Math.floor(Math.random()*destinations.length);
+  destinationDiv.classList.replace('load-destinations', 'filtered-destinations')
+  loadDestinations([destinations[randomDestination]]) 
+  setTimeout(()=> destinationDiv.innerHTML += `<div class="random-text">
+  <h4>Hey! This is a great spot for remote web developing, don't you think?</h4>
+  </div>
+  `, 700)
+}
+
+// Function for searchbar, returns destinations based on name, country and language.
+const handleSearch = () => {
+  const searchValue = searchInput.value.toLowerCase()
+    const searchResult = destinations.filter((dest) => {
+      return (
+        dest.name.toLowerCase().includes(searchValue) ||
+        dest.country.toLowerCase().includes(searchValue) ||
+        dest.language.toLowerCase().includes(searchValue)
+      )
+    })
+    loadDestinations(searchResult)
+  }
+  
+//EventListners 
+buttonRandom.addEventListener('click', filterRandom)
+buttonTemp.addEventListener('click', tempToggle)
+buttonRate.addEventListener('click', rateToggle)
+dropDown.addEventListener('change', filterCountry)
+searchInput.addEventListener('input', handleSearch)
```

Comment:
We think it is a great idea to have 'input' here instead of 'submit' so no need to click enter!
// Martin & Mai